source/hic_hal/target_config.h

```diff
@@ -74,6 +78,14 @@ typedef struct target_cfg {
     ram_region_info_t extra_ram[MAX_EXTRA_RAM_REGION + 1];      /*!< Extra RAM regions  */
     const char *rt_board_id;                                /*!< If assigned, this is a flexible board ID */
     uint16_t rt_family_id;                                     /*!< If assigned, this is a flexible board ID */
+#ifdef TRUSTZONE_M
+    uint32_t flash_start_alias;     /*!< Alias Address of the application entry point */
+    uint32_t flash_end_alias;       /*!< Alias Address where the flash ends */
+    uint32_t ram_start_alias;       /*!< Lowest contigous RAM Alias address the application uses */
+    uint32_t ram_end_alias;         /*!< Highest contigous RAM Alias address the application uses */
+    flash_region_info_t extra_flash_alias[MAX_EXTRA_FLASH_REGION + 1]; /*!< Alias Extra flash regions. */
+    ram_region_info_t extra_ram_alias[MAX_EXTRA_RAM_REGION + 1];       /*!< Alias Extra RAM regions. */
+#endif
```

Comment:
I'm not sure why the current api is not enough for the extra ram and flash

Reply:
v8M introduces secure/non secure alias for the same physical memory/location. extra ram and flash is for different ram and flash locations. musca_B will have two flashes and correcponding 4 s/ns alias

Reply:
Please use the existing extra regions instead of adding more. If needed, add a flags field to `flash_region_info_t` and `ram_region_info_t` structs.

The aliasing is not intrinsic to v8-M, it's simply a recommendation from Arm for how to implement the IDAU given the strict timing requirements.